src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,77 @@
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.springboot.datajpa.dto.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.ProductResponseDto;
+import mate.academy.springboot.datajpa.dto.mapper.ProductMapper;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+
+    public ProductController(ProductService productService, ProductMapper productMapper) {
+        this.productService = productService;
+        this.productMapper = productMapper;
+    }
+
+    @PostMapping
+    public ProductResponseDto create(@RequestBody ProductRequestDto requestDto) {
+        Product product = productService.save(productMapper.toProduct(requestDto));
+        return productMapper.toResponseDto(product);
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return productMapper.toResponseDto(productService.findById(id));
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteProduct(@PathVariable Long id) {
+        productService.deleteById(id);
+    }
+
+    @PutMapping("/update/{id}")
+    public ProductResponseDto updateProduct(@PathVariable Long id,
+                                            @RequestBody ProductRequestDto requestDto) {
+        Product product = productMapper.toProduct(requestDto);
+        product.setId(id);
+        return productMapper.toResponseDto(productService.update(product));
+    }
+
+    @GetMapping("/filter")
+    public List<ProductResponseDto> getAllByPrice(@RequestParam BigDecimal from,
+                                                  @RequestParam BigDecimal to) {
+        return productService.findAllByPriceBetween(from, to)
+                .stream()
+                .map(productMapper::toResponseDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping
+    public List<ProductResponseDto> getAllByCategories(
+            @RequestParam(required = false) List<Long> categoriesIds) {
+        List<Product> products;
+        if (categoriesIds != null) {
+            products = productService.findProductsByCategoriesIds(categoriesIds);
```

Comment:
It's better to move it to /filter endpoint. Or even better use only this endpoint, make all filter parameters optional and remove `filter` endpoint